simp_le.py

```diff
@@ -1183,5 +1210,33 @@ def test_error_exit_codes(self, dummy_stderr):
                 assert False
 
 
+class IOPluginTests(TestCase):
+    """Integration tests for plugin interface."""
+
+    # this is unittest suite | pylint: disable=missing-docstring
+
+    def setUp(self):
+        self.root = tempfile.mkdtemp()
+        self.key_path = os.path.join(self.root, 'key.pem')
+        test_key = os.path.join('testdata', "rsa2048_key.pem")
+        shutil.copy(test_key, self.key_path)
+
+    def tearDown(self):
+        shutil.rmtree(self.root)
+
+    def test_keyfile(self):
+        """Test whether newly saved key files are not world-readable."""
+        keyfile = IOPlugin.registered['key.pem']
+        keyfile.path = self.key_path
+
+        key_data = keyfile.load()
+        self.assertNotEqual(key_data, IOPlugin.EMPTY_DATA)
+        os.remove(self.key_path)
```

Comment:
if you do tests other way round: first gen, then save, then load, you don't have to do that `os.remove` as it's cleanup in `tearDown`